src/h5web/dataset-visualizer/DatasetVisualizer.tsx

```diff
@@ -31,7 +33,13 @@ function DatasetVisualizer(props: Props): JSX.Element {
       <div className={styles.displayArea}>
         {dataset ? (
           activeVis && (
-            <VisDisplay key={dataset.id} vis={activeVis} dataset={dataset} />
+            <VisProvider
+              key={dataset.id}
+              mapping={dimensionMapping}
+              dataset={dataset}
+            >
+              <VisDisplay key={dataset.id} vis={activeVis} dataset={dataset} />
```

Comment:
No need for a `key` here either way.

Reply:
I don't think you need to pass `dataset` here either any more.

Reply:
In the end, I took care of the remaining visualizations and removed `VisDisplay` altogether.